pkg/httpserver/server.go

```diff
@@ -155,51 +158,57 @@ func NewWithInterfaces(logger log.Logger, router *gin.Engine, tracer tracing.Ins
 		return nil, err
 	}
 
-	logger.Info("serving api requests on address %s", listener.Addr().String())
+	logger.Info("serving httpserver requests on address %s", listener.Addr().String())
 
 	apiServer := &HttpServer{
 		logger:   logger,
 		server:   server,
 		listener: listener,
+		settings: settings,
 	}
 
 	return apiServer, nil
 }
 
-func (a *HttpServer) Run(ctx context.Context) error {
-	go a.waitForStop(ctx)
+func (s *HttpServer) Run(ctx context.Context) error {
+	go s.waitForStop(ctx)
 
-	err := a.server.Serve(a.listener)
+	err := s.server.Serve(s.listener)
 
 	if !errors.Is(err, http.ErrServerClosed) {
-		a.logger.Error("Server closed unexpected: %w", err)
+		s.logger.Error("Server closed unexpected: %w", err)
 
 		return err
 	}
 
+	s.logger.Info("leaving httpserver")
+
 	return nil
 }
 
-func (a *HttpServer) waitForStop(ctx context.Context) {
+func (s *HttpServer) waitForStop(ctx context.Context) {
 	<-ctx.Done()
-	err := a.server.Close()
-	if err != nil {
-		a.logger.Error("Server Close: %w", err)
-	}
 
-	a.logger.Info("leaving api")
+	shutdownCtx, cancel := context.WithTimeout(context.Background(), s.settings.Timeout.Shutdown)
+	defer cancel()
+
+	s.logger.Info("trying to gracefully shutdown httpserver")
+
+	if err := s.server.Shutdown(shutdownCtx); err != nil {
+		s.logger.Error("Server Close: %w", err)
+	}
 }
 
-func (a *HttpServer) GetPort() (*int, error) {
-	if a == nil {
+func (s *HttpServer) GetPort() (*int, error) {
+	if s == nil {
 		return nil, errors.New("apiServer is nil, module is not yet booted")
```

Comment:
this log message looks really old, not only the mention of apiServer, but also the module being not yet booted

```suggestion
		return nil, errors.New("httpServer is nil, module is not yet running")
```